platform/mbed_error_log.h

```diff
@@ -0,0 +1,44 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2013 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_ERROR_LOG_H
+#define MBED_ERROR_LOG_H
+
+#ifndef MBED_CONF_ERROR_LOG_SIZE
+    #define MBED_CONF_ERROR_LOG_SIZE  5
+#else
+    #if MBED_CONF_ERROR_LOG_SIZE == 0
+        #define MBED_CONF_ERROR_LOG_SIZE  1
+    #endif    
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+MbedErrorStatus mbed_log_put_error(mbed_error_ctx *error_ctx);
```

Comment:
No docs, and they are public.

Reply:
Actually this is not intended to be public, we don't want this to be called directly, public APIs are the ones in mbed_error.h

Reply:
It's in the header, so I think even if consumer are other parts of Mbed OS (outside of file) we should get some simple docs.

Reply:
I see, let me see how we can address this. I was not thinking of exposing it outside. But Ill think about it and if its useful we can expose it.

Reply:
@deepikabhavnani  - Good catch, I'll update the doxygen comments.